plugins/modules/azure_rm_notificationhub.py

```diff
@@ -0,0 +1,408 @@
+          - Use C(present) to create or update an notification hub and C(absent) to delete it.
+      default: present
+      choices:
+          - absent
+          - present
+      type: str
+    log_path:
+        description:
+            - parent argument.
+        type: str
+    log_mode:
+        description:
+            - parent argument.
+        type: str    
+extends_documentation_fragment:
+    - azure.azcollection.azure
+    - azure.azcollection.azure_tags
+
+author:
+    - Praveen Ghuge (@praveenghuge)
+    - Karl Dasan (@karldas30)
+'''
+EXAMPLES = '''
+
+- name: "Create Notification Hub"
+  azure_rm_notificationhub:
+    resource_group: testgroupans
+    location: eastus
+    namespace_name: myNamespace
+    name: myhub
+    tags:
+       - a: b
+    sku: free
+
+- name: Delete Notification Hub
+  azure_rm_notificationhub:
+    resource_group: testgroupans
+    name: myNamespace
+    state: absent
+
+- name: "Create Notification Hub Namespace"
+  azure_rm_notificationhub:
+    resource_group: testgroupans
+    location: eastus
+    namespace_name: myNamespace
+    tags:
+       - a: b
+    sku: free
+
+- name: Delete Notification Hub Namespace
+  azure_rm_notificationhub:
+    resource_group: testgroupans
+    namespace_name: myNamespace
+    state: absent
+
+'''
+
+RETURN = '''
+state:
+    description:
+        - Current state of the Notification Hub namesapce or Notification Hub.
+    returned: always
+    type: dict
+    sample: {
+        "additional_properties": {},
+        "critical": false,
+        "data_center": null,
+        "enabled": true,
+        "location": "eastus2",
+        "metric_id": null,
+        "name": "testnaedd3d22d3w",
+        "namespace_type": "NotificationHub",
+        "provisioning_state": "Succeeded",
+        "region": null,
+        "scale_unit": null,
+        "service_bus_endpoint": "https://testnaedd3d22d3w.servicebus.windows.net:443/",
+        "sku": "Free",
+        "tags": {
+            "a": "b"
+        },
+        "type": "Microsoft.NotificationHubs/namespaces"
+    }    
+'''
+
+import time
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.mgmt.notificationhubs.models import NotificationHubCreateOrUpdateParameters, NamespaceCreateOrUpdateParameters
+    from azure.mgmt.notificationhubs.models.sku import Sku
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureNotificationHub(AzureRMModuleBase):
+
+    def __init__(self):
+        # define user inputs from playbook
+
+        self.authorizations_spec = dict(
+            name=dict(type='str', required=True)
+        )
+
+        self.module_arg_spec = dict(
+            resource_group=dict(type='str', required=True),
+            namespace_name=dict(type='str', required=True),
+            name=dict(type='str'),
```

Comment:
```suggestion
            name=dict(type='str', required=True),
```

Reply:
name is not required property, user can create just notification_hub_namespace